wrappers/python/simtk/openmm/app/pdbfile.py

```diff
@@ -370,8 +372,8 @@ def writeModel(topology, positions, file=sys.stdout, modelIndex=None, keepIds=Fa
                     else:
                         atomName = atom.name
                     coords = positions[posIndex]
-                    line = "%s%5d %-4s %3s %s%4s    %s%s%s  1.00  0.00          %2s  " % (
-                        recordName, atomIndex%100000, atomName, resName, chainName, resId, _format_83(coords[0]),
+                    line = "%s%5d %-4s %3s %s%4s%s   %s%s%s  1.00  0.00          %2s  " % (
+                        recordName, atomIndex%100000, atomName, resName, chainName, resId, resIC, _format_83(coords[0]),
```

Comment:
If `resIC` is more than 1 character long, this poops out a broken PDB file :(

Same is true if chainName is more than 1 character, resName is more than 3 characters, atomName is more than 4 characters, etc.

Might be worth adding some protection in here so it properly truncates all strings to at most the right length.

Reply:
In either of those cases, it throws an exception on the next line:

```python
assert len(line) == 80, 'Fixed width overflow detected'
```

This only happens if the caller specifies `keepIds=True`, and we do warn them it's their responsibility to make sure the IDs are valid!

I should probably change the format spec to say `%1s` for that field, though, so if `resIC` is an empty string it will work correctly.

Reply:
Another word of warning, then:  Python compiles pure-Python modules into bytecode (.pyc files now buried in `__pycache__` folders).  And it has 3 optimization levels (including no optimizations, the default)!  But they're all kinda... lame.

* `-O` goes through and trashes all `assert` statements
* `-OO` goes through and trashes all docstrings in addition to `assert` statements

The intent here according to email threads I've seen in Python user and developer forums is for `assert` to be strictly a debugging-like tool.  Spread them around liberally to catch bugs early.  You can then eliminate their performance hit in production by optimizing all `assert` statements out.

The resulting conventions for how `assert` should be used is:

1. *Never* tuck a side-effect into an `assert` statement.  Very bad:
```python
assert my_obj.initialize_and_verify(), "Verification failed"
```
2. As a general rule, don't use `assert` to check that user input is valid

After all, you never know when somebody will install your library with optimizations and wipe out all of your asserts.

Reply:
Good point.  I'll change it to throw an exception more properly.